### PR TITLE
(feat): Add metadata to alarms, notifications, and nps

### LINKED ISSUE #57: Include `metadata` in all create functions.
**Feature Request / Enhancement**

It appears that `metadata` is being excluded in several functions which *create* Resources. Here are a few examples:

```
create_notification()  # L637 in rackspace.py
create_notification_plan()  # L696 in rackspace.py
create_alarm()  # L517 in rackspace.py
```

It *is* implemented within the following resource creation functions:

```
create_entity()
create_check()
```

If it would be possible to update them all to accept metadata, that would be very helpful. My specific use-case is that I am using metadata as a means to associate specific resources with my application.

**Relevant Source File:** https://github.com/racker/rackspace-monitoring/blob/master/rackspace_monitoring/drivers/rackspace.py#L637

**Documentation demonstrating API accepts Metadata:** 
- https://developer.rackspace.com/docs/rackspace-monitoring/v1/api-reference/notifications-operations/
- https://developer.rackspace.com/docs/rackspace-monitoring/v1/api-reference/notification-plans-operations/
- https://developer.rackspace.com/docs/rackspace-monitoring/v1/api-reference/alarms-operations/

### PR DESCRIPTION
**Pull request by dimtruck**
* Adds metadata to create_alarm, create_notification, and create_notification_plan
* Add tests

closes #57

### CODE REVIEW

File: rackspace_monitoring/drivers/rackspace.py [hunk 1]
```diff
@@ -522,7 +522,8 @@ def create_alarm(self, entity, **kwargs):
                 'label': kwargs.get('label'),
                 'check_id': kwargs.get('check_id'),
                 'criteria': kwargs.get('criteria'),
-                'notification_plan_id': kwargs.get('notification_plan_id')}
+                'notification_plan_id': kwargs.get('notification_plan_id'),
+                'metadata': kwargs.get('extra', {})}
```
Comment (jjbuchan) [inline rackspace_monitoring/drivers/rackspace.py]: Should this be `metadata` instead of `extra`?

Comment (dimtruck) [inline rackspace_monitoring/drivers/rackspace.py]: makes me question the validity of these tests :/

**Comment by dimtruck** [comment]
@jjbuchan rfr

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+rackspace-monitoring v0.8.0 - 2017-08-10:
+
+* Add support for metadata in alarms, notifications, and notification plans
+* Fix null attribute filtering for create and
```
Comment (jjbuchan) [inline CHANGES.md]: One more thing.  It includes https://github.com/racker/rackspace-monitoring/pull/56 - `Add active/scheduled suppression fields to entity/check/alarm objects`

**Pull request merged**